src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -269,32 +279,144 @@ t8_forest_adapt (t8_forest_t forest)
+      /* We first assume that we have an (in)complete family. 
+       * In the following we try to disprove this by running 2 tests.
+       * */
+      if (0 < tscheme->t8_element_level(elements_from[0])) {
+        /* Assume family we are looking a family */
+        is_family = 1;
+
+        /* el_c is the Index of the el_considered in elements_from_copy */
+        if (num_el_from < (t8_locidx_t) num_children){
+          el_c = 0;
+        }
+        else {
+          el_c = num_children - zz;
+        }
+        /* If el_c == 0 then elements_from_copy is equal to elements_from */
+        for (zz = 0; zz < num_children &&
+                     el_considered + (t8_locidx_t) zz - (t8_locidx_t) el_c < num_el_from; zz++) {
+            elements_from_copy[zz] = t8_element_array_index_locidx (telements_from,
+                                                                    el_considered + zz - el_c);
+        }
+        
+        level_current = tscheme->t8_element_level(elements_from_copy[el_c]);
+        tscheme->t8_element_parent(elements_from_copy[el_c], element_parent_current);
+
+        /* Test 1: Check if already considered elements of current family passed,
+         * so current considered element can not get coarsed any more.
+         * */
+        for (zz = 1; zz < num_children && 
+                     el_considered - (t8_locidx_t) zz > -1; zz++)
+        {
+          tscheme->t8_element_parent (t8_element_array_index_locidx (telements_from,
+                                                                     el_considered - zz),
+                                      element_parent_compare);
+          if (!tscheme->t8_element_compare(element_parent_current, element_parent_compare)) {
+            is_family = 0;
+          }
+        }
+
+        /* Test 2: Check if elements in elements_from_copy get "eaten" by coarsing current element */
+        if (is_family) {
+          for (zz = 0; zz < num_children &&
+                       el_considered + (t8_locidx_t) zz - (t8_locidx_t) el_c < num_el_from; zz++) {
+            level = tscheme->t8_element_level(elements_from_copy[zz]);
+            /* Only elements with higher level then level of current element, can get eaten. */
+            if (level > level_current) {
+              tscheme->t8_element_copy(elements_from_copy[zz], element_parent_compare);
+              while(level > level_current-1) {
+                tscheme->t8_element_parent (element_parent_compare, element_parent_compare);
+                level = tscheme->t8_element_level(element_parent_compare);
+              }
+              if (!tscheme->t8_element_compare(element_parent_current, element_parent_compare)) {
+                is_family = 0;
+              }
+            }
+          }
+        }
+
+        /* Test 3: Check num_childrem elements before el_considered, if they will get eaten.
+         * Reason: current element could be on boarder
+         * [IL] Question: Is this required if el_c > 0
```

Comment:
Have you answered this question?

Reply:
Not yet, I have to look into the question more closely.

Reply:
Answer: It is not necessary.
I have extended the if statement.